src/transactions/reducer.ts

```diff
@@ -121,7 +126,10 @@ export const reducer = (
 
       return {
         ...state,
-        transactions: action.transactions,
+        transactions: {
+          ...state.transactions,
+          [action.networkId]: action.transactions,
```

Comment:
seems like an alternative to this whole refactor would have been to only overwrite the values with the same network id and leave the rest. I don't think this is more expensive because you're already copying the previous state which is linear complexity (same as a filter)

Reply:
oh interesting....can you expand on the benefits of that approach over this one? i think i'm open to it, but i'm not sure that i see the advantage aside from less code changes. one thing i like about the current approach is that we keep the "cached" data on the client the same as what it is on the server side (easier to debug, also easier visualise what changed in flipper - reordering the whole transactions array makes seeing the actual changes really difficult)

Reply:
I approved anyway because I don't think the benefit is large. 

One upside I can think of is that it keeps state more similar to what the user ends up seeing in their feed. It also limits the scope of the changes, like you mentioned, and might mean a bit less code to maintain. Given that we don't have a use case right now of showing transactions separately per blockchain, it doesn't seem necessary to store them separately, even if we do query for them separately. Given a choice between "matching what the user sees" and "matching how we fetch the data" I tend to opt for the former when choosing how to represent and store data. 

All that being said, you did it this way, and I think the difference between both options are minor enough to make it author's choice. Just wanted to voice the alternative while I was noticing it

Reply:
yes that makes sense, thanks for elaborating! i agree generally storing data in the format that the client needs to use it is the best option. with selectors we can get the best of both worlds here and hide the complexity in there before it reaches the display components while minimising the amount of data processing that goes into the cached data